lib/gollum/app.rb

```diff
@@ -179,7 +179,7 @@ def wiki_new
       halt 500 unless tempfile.is_a? Tempfile
 
       # Remove page file dir prefix from upload path if necessary -- committer handles this itself
-      dir      = wiki.per_page_uploads ? params[:upload_dest].match(/^(#{wiki.page_file_dir}\/+)?(.*)/)[2] : 'uploads'
+      dir      = wiki.per_page_uploads ? params[:upload_dest] : ::File.join([wiki.page_file_dir, 'uploads'].compact)
```

Comment:
I agree that `page_file_dir` should be completely internal, just as #1052 suggests. But further than that, I've almost always thought that Gollum is really sloppy with handling internal paths (several code regions seem to each handle them in a separate way) which logically resulted in many issues. If I remember correctly, I also encountered several places in the code that allow for external files (outside the given repository) to be displayed on the wiki. I tried to fix this in my forks but never finished it.

Personally, I think it's really time to revise all path-related features and come up with a unified, documented and enforced model...